src/Helpers/constants.js

```diff
@@ -667,6 +668,68 @@ export const VULNERABILITIES_HEADER = [
     }
 ];
 
+const groupsColumn = (isWorkspaceEnabled) => (
+    {
+        key: 'groups',
+        title: groupOrWorkspace(isWorkspaceEnabled),
+        inventoryGroupsFeatureFlag: true,
+        isShownByDefault: true,
+        renderFunc: (data, value, { inventory_group: group }) =>
+            <span>
+                {isEmpty(group) ?
+                    <span className="pf-v5-u-disabled-color-200">No {groupOrWorkspace(isWorkspaceEnabled).toLowerCase()}</span>
+                    : group[0].name} {/* currently, one group at maximum is supported */}
+            </span>
+    }
+);
+
+const tagsColumn = {
+    key: 'tags',
+    title: intl.formatMessage(messages.systemsColumnHeaderTags),
+    props: { isStatic: true },
+    isShownByDefault: true
+};
+
+const osColumn = {
+    key: 'os',
+    sortKey: 'os',
+    dataLabel: intl.formatMessage(messages.systemsColumnHeaderOS),
+    title: (
+        <Tooltip content={intl.formatMessage(messages.systemsColumnHeaderOSFull)}>
+            <span>
+                {intl.formatMessage(messages.systemsColumnHeaderOS)}
+            </span>
+        </Tooltip>
+    ),
+    /*eslint-disable camelcase*/
+    renderFunc: (osName, _, { rhsm_lock }) =>
+        rhsm_lock?.length ? <Tooltip
+            content={`Your RHEL version is locked at version ${rhsm_lock}`}
+        >
+            <Flex flex={{ default: 'inlineFlex' }}>
+                <FlexItem spacer={{ default: 'spacerSm' }}>{osName}</FlexItem>
+                <FlexItem spacer={{ default: 'spacerSm' }}>
+                    <InfoCircleIcon size="sm" color="var(--pf-v5-global--info-color--100)" />
+                </FlexItem>
+            </Flex>
+        </Tooltip>
+            : <span>{osName}</span>,
```

Comment:
If we extract this part into a single proper component, we can test the column in isolation, like any other component.

Reply:
You're so right!